client/src/test/java/io/spine/client/QueryFactoryTest.java

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package io.spine.client;
+
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Message;
+import com.google.protobuf.ProtocolStringList;
+import io.spine.protobuf.AnyPacker;
+import io.spine.test.client.TestEntity;
+import io.spine.test.client.TestEntityId;
+import io.spine.type.TypeUrl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newHashSet;
+import static io.spine.test.Verify.assertContainsAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author Alex Tymchenko
+ */
+@SuppressWarnings({"LocalVariableNamingConvention",
+        "DuplicateStringLiteralInspection"}) // A lot of similar test display names.
+@DisplayName("Query factory should")
+class QueryFactoryTest extends ActorRequestFactoryTest {
+
+    // See {@code client_requests} for declaration.
+    private static final Class<TestEntity> TARGET_ENTITY_CLASS = TestEntity.class;
+
+    @Nested
+    @DisplayName("when composing query")
+    class ComposeQueryTest {
+
+        @Test
+        @DisplayName("create proper `read all` query")
+        void composeReadAll() {
+            final Query readAllQuery = factory().query()
+                                                .all(TARGET_ENTITY_CLASS);
+            assertNotNull(readAllQuery);
+
+            checkTypeCorrectAndFiltersEmpty(TARGET_ENTITY_CLASS, readAllQuery);
+
+            checkFieldMaskEmpty(readAllQuery);
+        }
+
+        @Test
+        @DisplayName("create proper `read by ids` query")
```

Comment:
Can you please change `ids` to `IDs` in display names? That would read much more consistent.